apps/pay/components/Currency/currency-dropdown.tsx

```diff
@@ -15,19 +16,20 @@ export default function CurrencyDropdown({
   style?: React.CSSProperties
   showOnlyFlag?: boolean
 }) {
-  const router = useRouter()
+  const searchParams = useSearchParams()
+  const display = searchParams?.get("display")
+
   const { data: currencyData } = useCurrencyListQuery()
-  const [selectedDisplayCurrency, setSelectedDisplayCurrency] = React.useState(
-    router.query.display && typeof router.query.display === "string"
-      ? router.query.display
-      : localStorage.getItem("display") ?? "USD",
-  )
+  const [selectedDisplayCurrency, setSelectedDisplayCurrency] = React.useState("USD")
   const [isDropDownOpen, setIsDropDownOpen] = React.useState(false)
 
   useEffect(() => {
-    if (router.query?.display && typeof router.query.display === "string") {
-      setSelectedDisplayCurrency(router.query.display)
-    }
+    const newDisplay =
+      display && typeof display === "string"
+        ? display
+        : localStorage.getItem("display") ?? "USD"
```

Comment:
I think this should come in params because:
- this code is present elsewhere and not specifically related to a component
- this is business logic that we could unit test

Reply:
Unit test may be for another PR ?